app/src/main/java/com/infomaniak/swisstransfer/workers/UploadWorker.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.workers
+
+import android.content.Context
+import androidx.compose.runtime.Immutable
+import androidx.hilt.work.HiltWorker
+import androidx.work.*
+import com.infomaniak.multiplatform_swisstransfer.SwissTransferInjection
+import com.infomaniak.sentry.SentryLog
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.ImportLocalStorage
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.mapLatest
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@HiltWorker
+class UploadWorker @AssistedInject constructor(
+    @Assisted appContext: Context,
+    @Assisted params: WorkerParameters,
+    private val importLocalStorage: ImportLocalStorage,
+    private val swissTransferInjection: SwissTransferInjection,
+) : BaseCoroutineWorker(appContext, params) {
+
+    private val uploadManager inline get() = swissTransferInjection.uploadManager
+
+    private val fileChunkSizeManager by lazy {
+        FileChunkSizeManager(
+            chunkMinSize = EXPECTED_CHUNK_SIZE,
+            chunkMaxSize = EXPECTED_CHUNK_SIZE,
+            totalChunks = TOTAL_CHUNKS,
+        )
+    }
+    private val uploadFileTask by lazy {
+        UploadFileTask(uploadManager, fileChunkSizeManager)
+    }
+
+    private var totalUploadedBytes = 0L
+
+    override suspend fun launchWork(): Result {
+        SentryLog.i(TAG, "Work launched")
+
+        if (uploadManager.getUploadsCount() == 0L) {
+            SentryLog.w(TAG, "No upload pending")
+            return Result.failure()
+        }
+
+        val uploadSession = uploadManager.initUploadSession(recaptcha = "Recaptcha")!!
```

Comment:
Do we want to return a `Result.failure()` if the uploadSession is `null` ? Instead of crashing ?